### PR TITLE
For #4931 - Implement Shield Tracking protection data panel

### PR DESCRIPTION
**Pull request by mcarare**
Implement bottom sheet layout

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -715,4 +715,20 @@
 
     <!-- Label for the checkbox to send the tab crash report to Mozilla -->
     <string name="crash_report_send_crash_label">Send crash report to Mozilla</string>
+
+    <!-- Tracking protection bottom sheet -->
+    <string name="trackers_count_note">Trackers blocked since %s</string>
```
Comment (jonalmeida) [inline app/src/main/res/values/strings.xml:720]: Each string needs to add descriptive context for localizers.

File: app/src/main/res/drawable/ic_arrowhead_next.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (jonalmeida) [inline app/src/main/res/drawable/ic_arrowhead_next.xml:5]: We have [`mozac_ic_arrowhead_right`](https://github.com/mozilla-mobile/android-components/blob/main/components/ui/icons/src/main/res/drawable/mozac_ic_arrowhead_right.xml) in AC already.

File: app/src/main/res/layout/dialog_tracking_protection_sheet.xml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingStart="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    android:paddingEnd="@dimen/activity_horizontal_margin"
+    android:paddingBottom="@dimen/activity_vertical_margin">
+
+    <ImageView
+        android:id="@+id/site_favicon"
```
Comment (jonalmeida) [inline app/src/main/res/layout/dialog_tracking_protection_sheet.xml:17]: I don't think we need this according to [the designs](https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2139%3A57211) I see here.

**Review by jonalmeida — COMMENTED** [review]
This looks great! ✨ 

There is a difference in [the designs](https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2139%3A57211), so we should adjust our layout to follow them.

File: app/src/main/java/org/mozilla/focus/ext/Context.kt [hunk 4]
```diff
@@ -19,3 +20,19 @@ val Context.application: FocusApplication
  */
 val Context.components: Components
     get() = application.components
+
+/**
+ * Get the app install date.
+ */
+fun Context.installedDate(): String {
```
[the remark this answers is not in the document]
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/focus/ext/Context.kt:27]: Same for the case below.

File: app/src/main/res/drawable/ic_arrowhead_next.xml (hunk 2 above)
Comment (mcarare) [reply app/src/main/res/drawable/ic_arrowhead_next.xml:5]: Ok, I will add ui-icons to dependencies.

File: app/src/main/res/layout/dialog_tracking_protection_sheet.xml (hunk 3 above)
Comment (mcarare) [reply app/src/main/res/layout/dialog_tracking_protection_sheet.xml:17]: This is the latest design: https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2447%3A65181 @brampitoyo can confirm.

**Comment by mcarare** [comment]
> This looks great! ✨
> 
> There is a difference in [the designs](https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2139%3A57211), so we should adjust our layout to follow them.

This is the latest design. https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2447%3A65181
Bram can confirm, I have sent him screenshots and screen recording with the changes from this PR.

**Comment by mcarare** [comment]
I will also wait for the answer from Bram [here](https://github.com/mozilla-mobile/focus-android/issues/5091#issuecomment-896546824) for a final style on the text used.

**Review by jonalmeida — APPROVED** [review]
Since there are strings in this PR too, I'm okay with landing it as is, and fixing the layout in a future PR if needed by UX.

**Review by ionutbedregeanu — APPROVED** [review]

**Pull request merged**